src/components/table/column-picker/column-picker.component.ts

```diff
@@ -388,21 +396,30 @@ export interface ColumnPickerActionsContext {
 
 /** An interface representing a grouped item */
 export interface ColumnPickerGroupItem {
+    // The name of the group that this column belongs to.
     group?: string;
+    // The name of the column.
     name: string;
 }
 
 /** An interface representing settings of groups defined in ColumnPickerGroupItem objects */
 export interface ColumnPickerGroupSetting {
+    // The name of the group this setting object is related to.
     group: string;
+    // Defines if this group will be expanded on load. This is an optional property.
     initiallyExpanded?: boolean;
 }
 
 /** Interface representing a tree node item. This normalises data into one format */
 export class ColumnPickerTreeNode {
+    // The name of the column or group.
```

Comment:
@donna-marijne That style of comment is for functions only - well at least according to the linter in VS code. My understanding was that we wanted to document the code as much as possible for the end user's benefit. That would mean that I'd only be able to comment above the function and have remove all the rest. Please let me know you thoughts.

Reply:
Which linter are you using? There is absolutely no rule that restricts doc-comments to functions. In this file there are dozens of doc comments on properties.

```
/** A tree-friendly representation of the deselected data */
_treeData: ColumnPickerTreeNode[];
```

> That would mean that I'd only be able to comment above the function and have remove all the rest.

Could you rephrase this because I'm not sure what it means?

Reply:
@donna-marijne  The code snippet shown was confusing I thought you meant inner comments but I see I used `//` on one of the function comments. Fixed now, good spot!

Reply:
To be clear, I am talking about this and other class/interface properties, not functions.